pkg/api/types.go

```diff
@@ -748,30 +748,55 @@ func (p *Properties) K8sOrchestratorName() string {
 	return ""
 }
 
-func (p *Properties) getAgentVMPrefix() string {
-	return p.K8sOrchestratorName() + "-agentpool-" + p.GetClusterID() + "-"
+func (p *Properties) getAgentPoolIndexByName(name string) int {
+	index := -1
+	for i, profile := range p.AgentPoolProfiles {
+		if profile.Name == name {
+			index = i
+			break
+		}
+	}
+	return index
+}
+
+// GetAgentVMPrefix returns the VM prefix for an agentpool
+func (p *Properties) GetAgentVMPrefix(a *AgentPoolProfile) string {
```

Comment:
@tariq1890 the VM prefix depends on more than just the name so we need to pass in the whole agent pool profile. I've fixed the comment to better reflect what the function is doing (the previous one was a leftover from an iteration where I passed in the index)

Reply:
Sounds good. Thanks!